### ISSUE TITLE
Not sure "no-unused-vars" should be triggered in this case

### ISSUE THREAD

**Issue by kud** [eslint/eslint#9573]
I've got this code:

```javascript
  check = () => {
    userService.isAuth(this.props.store)
  } 

 render() {
    const { component: Component, store, storeState, ...rest } = this.props

    return (
      <Route
        {...rest}
```

That's right, I don't use `store` in the `render` method. However, I need to extract it from props, to have `rest` without it, and I also use `store` in `check` method so `store` is necessarily given as props.

So I wonder if eslint should trigger or not here about `no-unused-vars`, what do you think?

**Comment by not-an-aardvark** [comment]
You can use the [`ignoreRestSiblings` option](https://eslint.org/docs/rules/no-unused-vars#ignorerestsiblings) to avoid warning on that case.

**Comment by kud** [comment]
Lovely, it works like a charm. Thank you very much @not-an-aardvark .
